// wechat-mp-hack/dist/util/request.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _fs = require('fs');

var _fs2 = _interopRequireDefault(_fs);

var _path = require('path');

var _path2 = _interopRequireDefault(_path);

var _request = require('request');

var _request2 = _interopRequireDefault(_request);

var _toughCookieFilestore = require('tough-cookie-filestore');

var _toughCookieFilestore2 = _interopRequireDefault(_toughCookieFilestore);

var _config = require('./config');

var _config2 = _interopRequireDefault(_config);

var _log = require('./log');

var _log2 = _interopRequireDefault(_log);

var _cache = require('./cache');

var _cache2 = _interopRequireDefault(_cache);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const cookieCache = new _cache2.default('cookie');
const j = _request2.default.jar(new _toughCookieFilestore2.default(cookieCache.cacheFile));
const r = _request2.default.defaults({
    method: 'POST',
    headers: {
        'Referer': _config2.default.baseurl,
        'Host': _config2.default.host,
        'User-Agent': _config2.default.userAgent,
        'X-Requested-With': 'XMLHttpRequest'
    },
    json: true,
    jar: j,
    qs: {
        lang: 'zh_CN'
    },
    followAllRedirects: true,
    followOriginalHttpMethod: true
});

const WechatRequest = options => {
    return new Promise((resolve, reject) => {
        r(options, (e, r, body) => {
            if (e) {
                reject(e);
                _log2.default.error(e);
            } else {
                resolve(body);
            }
        });
    });
};

WechatRequest.get = (...options) => {
    return r.get(...options);
};

WechatRequest.getJSON = (url, options) => {
    return WechatRequest(Object.assign({
        method: 'GET',
        url: url,
        // qs: {
        //     f: 'json',
        //     ajax: 1
        // }
    }, options));
};

WechatRequest.cookies = () => {
    let cookies = j.getCookies(_config2.default.baseurl);
    let obj = {};
    cookies.forEach(c => {
        obj[c.key] = c.value;
    });
   ;
    return obj;
};

exports.default = WechatRequest;
module.exports = exports['default'];
